src/main/java/no/idporten/userservice/api/ApiUserService.java

```diff
@@ -75,6 +73,29 @@ public UserResource updateUserAttributes(String id, UpdateAttributesRequest upda
         return convert(idPortenUser);
     }
 
+    @Transactional
+    public UserResource updateUserPidAttributes(UpdatePidAttributesRequest request) {
+        IDPortenUser user = userService.findFirstUser(request.getPersonIdentifier());
+        String closedCode = StringUtils.hasText(request.getClosedCode()) ? request.getClosedCode() : null;
+        List<String> helpDeskCaseReference = CollectionUtils.isEmpty(request.getHelpDeskReferences()) ? (new ArrayList<>()) : request.getHelpDeskReferences();
+
+        if (user == null) {
+            user = IDPortenUser.builder().pid(request.getPersonIdentifier()).helpDeskCaseReferences(helpDeskCaseReference).build();
+            user = userService.createStatusUser(user);
+        } else {
+            if (request.getHelpDeskReferences() != null) {
+                user.setHelpDeskCaseReferences(helpDeskCaseReference);
+            }
+        }
+
+        if (request.getClosedCode() != null) {
+            setStatus(user, closedCode);
```

Comment:
Her forstår jeg ikke helt flyten 😅 Vil ikke no.idporten.userservice.api.ApiUserService#setStatus på user-objektet mellom linje 83 og 84 fikse det samme, og spare en ekstra db-operasjon?

Reply:
Sant det. 
Legge closedCode inn i builder for ikke eksisterende, og flytte if'en som det er snakk om inn i else (altså om brukeren eksisterer)? Bruker setStatus for å også oppdatere Active og ClosedCodeLastUpdated.

Reply:
Det kan i alle fall fiksast på linje 83-84 men må leggast inn i user-objektet før det vert oppretta